code-samples/serving/cloudevents/cloudevents-rust/Dockerfile

```diff
@@ -11,10 +11,29 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+FROM rust:1.31.0 AS builder
```

Comment:
probably worth to take a look at https://github.com/knative/docs/pull/5757 as well.

Reply:
Sorry for the late reply!
I'm following the PR above as you suggested and the build was successful, but I get an image size of 2.32GB. I also tried to run and curl it but I got `curl: (56) Recv failure: Connection reset by peer` error.

```Dockerfile
FROM rust:1.73.0 AS builder

ARG USER=appuser
ARG USER_UID=1001
ARG USER_GID=$USER_UID

# Add a user so the server will run as a non-root user.
RUN addgroup --gid $USER_GID $USER && \
    adduser -u $USER_UID --ingroup $USER --disabled-password $USER

COPY . .

RUN cargo install --path .

USER $USER

CMD ["knative-cloudevents-example"]
```

Now as a first step I'm trying to reduce the image size by using a second step where I copy and execute the release.